rush-plugins/rush-buildxl-graph-plugin/src/test/GraphProcessor.test.ts

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import type { IOperationRunner, Operation } from '@rushstack/rush-sdk';
+import type { ShellOperationRunner } from '@rushstack/rush-sdk/lib/logic/operations/ShellOperationRunner';
+import { Terminal, NoOpTerminalProvider } from '@rushstack/terminal';
+
+import { GraphProcessor, type IGraphNode } from '../GraphProcessor';
+
+// to update the examples folder, run the following command from the project root:
+// export DEBUG_RUSH_BUILD_GRAPH="test" && rush build --production -t . --drop-graph ./src/examples/graph.json
+import graph from '../examples/graph.json';
+import debugGraph from '../examples/debug-graph.json';
+
+describe(GraphProcessor.name, () => {
+  let exampleGraph: readonly IGraphNode[];
+  let graphParser: GraphProcessor;
+  let emittedErrors: Error[];
+  let emittedWarnings: Error[];
+
+  beforeAll(() => {
+    exampleGraph = Array.from(graph.nodes).sort((a, b) => a.id.localeCompare(b.id));
```

Comment:
Don't use localeCompare in tests without specifying a locale. It is environment dependent. Use `<` and `>` instead.